docs/gitlab_integration.md

```diff
@@ -0,0 +1,132 @@
+# 1. Install/register GitLab Runner
+GitLab Runner is the open source project that is used to run your jobs and
+send the results back to GitLab. You have to
+[install](https://docs.gitlab.com/runner/install/) it on a machine where
+your analysis will be performed.
+
+Once GitLab Runner is installed, you need to register it with GitLab.
+You can use the following command:
+```sh
+sudo gitlab-runner register \
+    --non-interactive \
+    --url "https://mycompany.gitlab.com" \
+    --registration-token "$REGISTRATION_TOKEN" \
+    --description "codechecker" \
+    --executor "shell"
+```
+
+The token can be found in your projects settings under
+`settings/ci_cd/Set up a specific Runner manually`. You can find more
+information how to
+[register a GitLab Runner](https://docs.gitlab.com/runner/register/index.html).
+
+# 2. Define a job in your .gitlab-ci.yml
+GitLab CI/CD pipelines are configured using a YAML file called `.gitlab-ci.yml`
+within each project. Create this file in your project root directory:
+
+```yml
+code_quality:
+  allow_failure: false
+  script:
+    - bash .gitlab/ci/run_codechecker.sh
+  after_script:
+    - cat gl-code-quality-report.json
+  artifacts:
+    reports:
+      codequality: gl-code-quality-report.json
+    paths: [gl-code-quality-report.json]
+    expire_in: '5 yrs'
+  stage: test
+```
+
+`expire_in` allows you to specify how long artifacts should live before they
+expire and are therefore deleted. You can set it to a lower or a higher value.
+For more information
+[see](https://docs.gitlab.com/ee/ci/yaml/README.html#artifactsexpire_in).
+
+Script is a shell script which will be executed by the Runner.
+
+# 3. Create script to run CodeChecker analysis
+Create a shell script in your project under `.gitlab/ci/run_codechecker.sh`.
+This script will run the analysis and it has to create a json file called
+`gl-code-quality-report.json` which will contains reports in Code Climate
+format. CodeChecker from version `6.12.0` is able to generate Code Climate
+output by using the *CodeChecker parse* or *CodeChecker cmd diff* commands.
+
+```sh
+#!/usr/bin/env bash
+
+# Log your project.
+CodeChecker log -b "make" -o ./compilation_database.json
+
+# Analyze your project.
+CodeChecker analyze --ctu \
+  -o ./reports \
+  ./compilation_database.json
+
+# Create the report file by using the CodeChecker parse command.
+CodeChecker parse \
+  --trim-path-prefix $(pwd) \
+  -e codeclimate
+  ./reports > gl-code-quality-report.json
```

Comment:
Here the output is processed through a pipe but in the case of the diff example a file is created, which one is the final?

Reply:
I'm thinking to create two small yml examples and do not mix the two. One for the parse which which will parse and report all the found results back to gitlab, and another with the diff, because there a separate server with some runs is needed to what the comparison is done.

Reply:
See my comment here: https://github.com/Ericsson/codechecker/pull/2687#discussion_r412854244. I agree that it can annoying that at the parse command the results will be written to standard output, at the diff command in case of json the result will be printed to the standard output but in case of gerrit or codeclimate format the results will be written to a file. What do you recommend?

Reply:
After the discussion in the other thread we keep the json file for codeclimate and only dump the json output to the stdout.
In this case example shouldn't be just a file rename something like this?
```sh
CodeChecker parse \
  --trim-path-prefix $(pwd) \
  -e codeclimate
  -o parse_output
  ./reports

mv parse_output/codeclimate.json gl-code-quality-report.json
```